staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -17,49 +17,46 @@ limitations under the License.
 package logs
 
 import (
-	"github.com/go-logr/logr"
 	"github.com/spf13/pflag"
 
 	"k8s.io/klog/v2"
 
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/component-base/config"
 	"k8s.io/component-base/logs/sanitization"
 )
 
 // Options has klog format parameters
 type Options struct {
-	LogFormat       string
-	LogSanitization bool
+	Config config.LoggingConfiguration
 }
 
 // NewOptions return new klog options
 func NewOptions() *Options {
-	return &Options{
-		LogFormat: DefaultLogFormat,
-	}
+	return &Options{Config: config.LoggingConfiguration{Format: DefaultLogFormat}}
 }
 
 // Validate verifies if any unsupported flag is set
 // for non-default logging format
 func (o *Options) Validate() []error {
-	return ValidateLoggingConfiguration(o)
+	errs := ValidateLoggingConfiguration(&o.Config, field.NewPath("loggingConfiguration"))
```

Comment:
This should take the fieldpath to use as an argument... it's up to the consuming component what the user-facing path is. In the kubelet's case, it is `logging`:

```
Logging componentbaseconfigv1alpha1.LoggingConfiguration `json:"logging,omitempty"`
```

Reply:
ValidateLoggingConfiguration does take a fieldpath argument as it config based setup. Options on the other side are used for command flag based setup, in that case fieldpath doesn't make sense as flags are flat.

Reply:
if this is for CLI-flags only, then don't pass in a non-existent `loggingConfiguration` field path

Reply:
Done